test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -543,12 +567,241 @@ public void AnomalyDetectionWithSrCnn()
             int k = 0;
             foreach (var prediction in predictionColumn)
             {
+                Assert.Equal(3, prediction.Prediction.Length);
+                if (k == 20)
+                    Assert.Equal(1, prediction.Prediction[0]);
+                else
+                    Assert.Equal(0, prediction.Prediction[0]);
+                k += 1;
+            }
+        }
+
+        [Fact]
+        public void AnomalyDetectionWithSrCnnEntireAnomalyOnly()
+        {
```

Comment:
Are the three tests below nearly identical except for the mode?
If so, can you please refactor them using [Theory] instead of [Fact]?

Reply:
> Are the three tests below nearly identical except for the mode?
> If so, can you please refactor them using [Theory] instead of [Fact]?

Updated as you suggested.